lang/java/reef-runtime-azbatch/src/main/java/org/apache/reef/runime/azbatch/evaluator/EvaluatorShim.java

```diff
@@ -0,0 +1,215 @@
+
+  private final ExecutorService evaluatorLaunchExecutorService;
+
+  private Process evaluatorProcess;
+  private Integer evaluatorProcessExitValue;
+
+  @Inject
+  EvaluatorShim(final REEFFileNames fileNames,
+                final ConfigurationSerializer configurationSerializer,
+                final RemoteManager remoteManager,
+                @Parameter(DriverRemoteIdentifier.class)
+                final String driverRemoteId,
+                @Parameter(ContainerIdentifier.class)
+                final String containerId) {
+    this.fileNames = fileNames;
+    this.configurationSerializer = configurationSerializer;
+
+    this.driverRemoteId = driverRemoteId;
+    this.containerId = containerId;
+
+    this.remoteManager = remoteManager;
+    this.evaluatorShimStatusChannel = this.remoteManager.getHandler(this.driverRemoteId,
+        EvaluatorShimProtocol.EvaluatorShimStatusProto.class);
+
+    this.evaluatorShimCommandChannel = this.remoteManager
+        .registerHandler(EvaluatorShimProtocol.EvaluatorShimControlProto.class, this);
+
+    this.evaluatorLaunchExecutorService = Executors.newSingleThreadExecutor();
+  }
+
+  public void run() {
+    LOG.log(Level.FINEST, "Entering EvaluatorShim.run().");
+    this.onStart();
+  }
+
+  public void stop() {
+    this.onStop();
+  }
+
+  @Override
+  public void onNext(final RemoteMessage<EvaluatorShimProtocol.EvaluatorShimControlProto> remoteMessage) {
+    final EvaluatorShimProtocol.EvaluatorShimCommand command = remoteMessage.getMessage().getCommand();
+    if (command == EvaluatorShimProtocol.EvaluatorShimCommand.EVALUATOR_LAUNCH) {
+      LOG.log(Level.INFO, "Received a command to launch the Evaluator.");
+      EvaluatorShim.this.onEvaluatorLaunch(remoteMessage.getMessage().getEvaluatorLaunchCommand(),
+          remoteMessage.getMessage().getEvaluatorConfigString());
+    } else if (command == EvaluatorShimProtocol.EvaluatorShimCommand.TERMINATE) {
+      LOG.log(Level.INFO, "Received a command to terminate.");
+      EvaluatorShim.this.onStop();
+    } else {
+      LOG.log(Level.WARNING, "An unknown command was received by the EvaluatorShim: {0}.",
+          command);
+      throw new IllegalArgumentException("An unknown command was received by the EvaluatorShim.");
+    }
+  }
+
+  private void onStart() {
+    LOG.log(Level.FINEST, "Entering EvaluatorShim.onStart().");
+
+    LOG.log(Level.INFO, "Reporting back to the driver with Shim Status = {0}",
+        EvaluatorShimProtocol.EvaluatorShimStatus.ONLINE);
+    this.evaluatorShimStatusChannel.onNext(
+        EvaluatorShimProtocol.EvaluatorShimStatusProto
+            .newBuilder()
+            .setRemoteIdentifier(this.remoteManager.getMyIdentifier())
+            .setContainerId(this.containerId)
+            .setStatus(EvaluatorShimProtocol.EvaluatorShimStatus.ONLINE)
+            .build());
+  }
+
+  private void onStop() {
+    LOG.log(Level.FINER, "Entering EvaluatorShim.onStop().");
```

Comment:
onStart() is using level FINEST. Here is using FINER. It'd better to make sure they will show in the same level log.

Reply:
changed.